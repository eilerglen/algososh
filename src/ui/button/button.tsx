import React from "react";
import styles from "./button.module.css";
import loaderIcon from "../../images/icons/loader.svg";
import { Ascending } from "../../ui/icons/ascending/ascending";
import { Descending } from "../../ui/icons/descending/descending";
import  {Direction}  from "../../types/enums/direction";

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  text?: string;
  type?: "button" | "submit" | "reset";
  sorting?: Direction;
  linkedList?: "small" | "big";
  isLoader?: boolean;
  mixin?: string;
}

export const Button: React.FC<ButtonProps> = ({
  text,
  mixin = "",
  type = "button",
  isLoader = false,
  sorting,
  linkedList,
  disabled,
  ...rest
}) => {
  const currentIcon =
    sorting === "ascending" ? <Ascending /> : <Descending />;
  const className = `text text_type_button text_color_primary ${
    styles.button
  } ${linkedList && styles[linkedList]} ${
    isLoader && styles.loader
  } ${mixin}`;

  return (
    <button
      className={className}
      type={type}
      disabled={isLoader || disabled}
      {...rest}
    >
      {isLoader ? (
        <img className={styles.loader_icon} src={loaderIcon} alt="Загрузка." />
      ) : (
        <>
          {sorting && currentIcon}
          <p className={`text ${sorting && "ml-5"}`}>{text}</p>
        </>
      )}
    </button>
  );
};
