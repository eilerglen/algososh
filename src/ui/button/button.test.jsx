import { Button } from "./button";
import renderer from 'react-test-renderer';
import {render, fireEvent, waitFor, screen} from "@testing-library/react"

const buttonText = "text"

describe("Check render button",() => {
    it("Checking button with text render", ()=> {
        const snap = renderer.create(<Button text="test text"/>).toJSON()
        expect(snap).toMatchSnapshot()
    });

    it("Checking button without text render", ()=> {
        const snap = renderer.create(<Button />).toJSON()
        expect(snap).toMatchSnapshot()
    });

    
    it("Checking button disabled render", ()=> {
        const snap = renderer.create(<Button disabled = {true} />).toJSON()
        expect(snap).toMatchSnapshot()
    });
    it("Checking button disabled render", ()=> {
        const snap = renderer.create(<Button isLoader = {true} />).toJSON()
        expect(snap).toMatchSnapshot()
    });

    it("Checking button click", () => {
        const fn = jest.fn()
        render(<Button text = {buttonText} onClick={fn}/>)
        const button = screen.getByText(buttonText)
        fireEvent.click(button)
    })

})
