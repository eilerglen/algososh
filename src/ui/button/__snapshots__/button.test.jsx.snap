// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render button Button disabled render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check render button Button disabled render 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Check render button Button with text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`Check render button Button without text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check render button Checking button disabled render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check render button Checking button disabled render 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Check render button Checking button with text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`Check render button Checking button without text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
